### PR TITLE
6999-CWPKI0023E-commit1

### LINKED ISSUE #6999: CWPKI0023E CWPKI0024E Certificate not found message improvements.
Messages CWPKI0023E and CWPKI0024E can be issued if the target certificate is invalid, expired. That should be clarified in the messages.

### PR DESCRIPTION
**Pull request by dburdine**
Messages CWPKI0023E and CWPKI0024E can be issued if the target certificate is invalid, expired. That should be clarified in the messages.

fixes #6999

### CODE REVIEW

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 1]
```diff
@@ -117,13 +117,13 @@ ssl.client.handshake.error.CWPKI0022E=CWPKI0022E: SSL HANDSHAKE FAILURE:  A sign
...
 ssl.client.handshake.error.CWPKI0022E.useraction=Review the extended error message from the TrustManager to determine what needs to change between the target SSL configuration and the client SSL configuration.
 # -------------------------------------------------------------------------------------------------
-ssl.client.alias.not.found.CWPKI0023E=CWPKI0023E: The certificate alias {0} specified by the property com.ibm.ssl.keyStoreClientAlias is not found in KeyStore {1}.
-ssl.client.alias.not.found.CWPKI0023E.explanation=The certificate alias specified for this SSL configuration is not in the specified KeyStore. 
-ssl.client.alias.not.found.CWPKI0023E.useraction=Either add a certificate into the KeyStore with the specified certificate alias or change the specified certificate alias to match an alias found in the client KeyStore.
+ssl.client.alias.not.found.CWPKI0023E=CWPKI0023E: The certificate alias {0} specified by the property com.ibm.ssl.keyStoreClientAlias is either not found in KeyStore {1} or it is invalid.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: ok

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 2]
```diff
@@ -117,13 +117,13 @@ ssl.client.handshake.error.CWPKI0022E=CWPKI0022E: SSL HANDSHAKE FAILURE:  A sign
...
+ssl.client.alias.not.found.CWPKI0023E.explanation=The certificate alias specified for this SSL configuration is not in the specified KeyStore or it was found to be invalid. An expired certificate is one example of an invalid certificate.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: ok

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 3]
```diff
@@ -117,13 +117,13 @@ ssl.client.handshake.error.CWPKI0022E=CWPKI0022E: SSL HANDSHAKE FAILURE:  A sign
...
+ssl.client.alias.not.found.CWPKI0023E.useraction=Either add a certificate into the KeyStore with the specified certificate alias, or replace or renew the invalid certificate with the specified certificate alias found in the client KeyStore.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Are "KeyStore" and "client KeyStore" the same thing?
Is "certificate" the same as "certificate alias" mentioned in the Message Text and .explanation? Terms should be consistent throughout.

Is there any way the user could tell whether the certificate was not found, invalid or expired?

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 4]
```diff
@@ -117,13 +117,13 @@ ssl.client.handshake.error.CWPKI0022E=CWPKI0022E: SSL HANDSHAKE FAILURE:  A sign
...
 # -------------------------------------------------------------------------------------------------
-ssl.server.alias.not.found.CWPKI0024E=CWPKI0024E: The certificate alias {0} specified by the property com.ibm.ssl.keyStoreServerAlias is not found in KeyStore {1}.
-ssl.server.alias.not.found.CWPKI0024E.explanation=The certificate alias specified for this SSL configuration is not in the specified KeyStore. 
-ssl.server.alias.not.found.CWPKI0024E.useraction=Either add a certificate into the KeyStore with the specified certificate alias or change the specified certificate alias to match an alias found in the server KeyStore.
+ssl.server.alias.not.found.CWPKI0024E=CWPKI0024E: The certificate alias {0} specified by the property com.ibm.ssl.keyStoreServerAlias is either not found in KeyStore {1} or it is invalid.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: ok

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 5]
```diff
@@ -117,13 +117,13 @@ ssl.client.handshake.error.CWPKI0022E=CWPKI0022E: SSL HANDSHAKE FAILURE:  A sign
...
+ssl.server.alias.not.found.CWPKI0024E.explanation=The certificate alias specified for this SSL configuration is not in the specified KeyStore or it was found to be invalid. An expired certificate is one example of an invalid certificate.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: ok

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 6]
```diff
@@ -117,13 +117,13 @@ ssl.client.handshake.error.CWPKI0022E=CWPKI0022E: SSL HANDSHAKE FAILURE:  A sign
...
+ssl.server.alias.not.found.CWPKI0024E.useraction=Either add a certificate into the KeyStore with the specified certificate alias, or replace, or renew the invalid certificate with the specified certificate alias found in the server KeyStore.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: see comments on CWPKI0023E, line 122 above

**Comment by skasund** [comment]
L2 message review completed

**Comment by dmuelle** [comment]
ID standing by on this one, per @dburdine, to see if this defect is necessary

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 1 above)
Comment (helyarp) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Hi, please use the "the _variable_ _noun_" format. Thus, the message should be:

The {0} certificate alias specified by the com.ibm.ssl.keyStoreClientAlias property is either not found in the {1} KeyStore or it is invalid.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 3 above)
Comment (helyarp) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: I think this useraction is okay. If you want to improve it, clarify how the "the KeyStore" is different from "the client KeyStore" as David mentions.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 4 above)
Comment (helyarp) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Use the "the _variable_ _noun_" format:
The {0} certificate alias specified by the com.ibm.ssl.keyStoreServerAlias property is either not found in the {1} KeyStore or it is invalid.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 5 above)
Comment (helyarp) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: okay

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 6 above)
Comment (helyarp) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Remove comma before "or renew":
... the specified certificate alias, or replace or renew the invalid certificate with the specified certificate alias ...

**Pull request merged**